src/app/formoverview/FormOverviewApiClient.ts

```diff
@@ -0,0 +1,130 @@
+import axios, { AxiosError, AxiosInstance } from 'axios';
+
+export interface FormOverviewApiClientProps {
+  /**
+   * The API key to authenticate requests
+   */
+  apiKey: string;
+  /**
+   * API base URL without ending /
+   */
+  baseUrl: string;
+  /**
+   * The header that contains the api key
+   * @default x-api-key
+   */
+  apiHeader?: string;
+  /**
+   * Optionally provide an axios instance
+   * @default - a new instance will be created
+   */
+  axios?: AxiosInstance;
+  /**
+   * timeout in ms
+   * @default 2000
+   */
+  timeout?: number;
+}
+
+export class FormOverviewApiClient {
+  private props: FormOverviewApiClientProps;
+  private axios: AxiosInstance;
+  private timeout: number;
+
+  /**
+   * Connects to API's. Use .post() or .get() to get the actual info
+   */
+  constructor(props: FormOverviewApiClientProps) {
+    this.props = props;
+    this.axios = this.initAxios({ axiosInstance: props.axios });
+    this.timeout = props.timeout ?? 2000;
+  }
+
+  private initAxios(config: {
+    axiosInstance?: AxiosInstance | undefined;
+  }): AxiosInstance {
+    if (config.axiosInstance) {
+      return config.axiosInstance;
+    } else {
+      return axios.create({
+        baseURL: this.props.baseUrl,
+      });
+    }
+  }
+
+  setTimeout(timeout: number) {
+    this.timeout = timeout;
+  }
+
+  async postData(endpoint: string, body: any, headers?: any): Promise<any> {
+    const newHeaders = this.addApiKeyHeader(this.props.apiKey, this.props.apiHeader, headers);
```

Comment:
kun je deze niet in de axios config opnemen?

Reply:
Ik heb voorlopig een kopie gemaakt met het idee om het straks generieker te doen. Ook met de generieke Apiclient uit de modules indien mogelijk.